src/panels/config/integrations/integration-panels/bluetooth/bluetooth-device-page.ts

```diff
@@ -0,0 +1,184 @@
+
+  @property({ attribute: false }) public route!: Route;
+
+  @property({ type: Boolean }) public narrow = false;
+
+  @property({ attribute: "is-wide", type: Boolean }) public isWide = false;
+
+  @state() private _rows: { [id: string]: DeviceRowData } = {};
+
+  private _unsub?: Promise<UnsubscribeFunc>;
+
+  private _firstUpdatedCalled = false;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.hass && this._firstUpdatedCalled) {
+      this._unsub = subscribeBluetoothAdvertisements(
+        this.hass,
+        this._handleIncomingEventDataMessage.bind(this)
+      );
+    }
+  }
+
+  protected firstUpdated(changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    if (this.hass) {
+      this._unsub = subscribeBluetoothAdvertisements(
+        this.hass,
+        this._handleIncomingEventDataMessage.bind(this)
+      );
+      this._firstUpdatedCalled = true;
+    }
+  }
+
+  private _handleIncomingEventDataMessage(event: EventDataMessage) {
+    if (event.add) {
+      for (const device_data of event.add) {
+        this._rows[device_data.address] = device_data;
+      }
+    }
+    if (event.change) {
+      for (const device_data of event.change) {
+        this._rows[device_data.address] = device_data;
+      }
+    }
+    if (event.remove) {
+      for (const device_data of event.remove) {
+        delete this._rows[device_data.address];
+      }
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._unsub) {
+      this._unsub.then((unsub) => unsub());
+      this._unsub = undefined;
+    }
+  }
+
+  private _columns = memoizeOne(
+    (localize: LocalizeFunc): DataTableColumnContainer => {
+      const columns: DataTableColumnContainer<DeviceRowData> = {
+        address: {
+          title: localize("ui.panel.config.bluetooth.address"),
+          sortable: true,
+          filterable: true,
+          showNarrow: true,
+          main: true,
+          hideable: false,
+          moveable: false,
+          direction: "asc",
+          flex: 2,
+        },
+        name: {
+          title: localize("ui.panel.config.bluetooth.name"),
+          filterable: true,
+          sortable: true,
+        },
+        source: {
+          title: localize("ui.panel.config.bluetooth.source"),
+          filterable: true,
+          sortable: true,
+        },
+        rssi: {
+          title: localize("ui.panel.config.bluetooth.rssi"),
+          type: "numeric",
+          sortable: true,
+        },
+      };
+
+      return columns;
+    }
+  );
+
+  protected render(): TemplateResult {
+    return html`
+      <hass-tabs-subpage-data-table
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        .route=${this.route}
+        .columns=${this._columns(this.hass.localize)}
+        .data=${Object.values(this._rows).map((row) => ({
+          ...row,
+          id: row.address,
+        }))}
```

Comment:
Move this to a `memoizeOne` so we don't re-render the table it on every state change